packages/validator/src/utils/utils.ts

```diff
@@ -0,0 +1,332 @@
+import * as logger from './logger';
+import { Log, Roots } from '../types';
+
+/**
+ * Replaces all "__match" instances in provided object and all it's children, except children of "__conditions"
+ * @param match - string that "__match" instances will be replaced with
+ * @param template - object in which "__match" instances will be replaced in
+ * @returns specs object with replaced "__match" instances
+ */
+export function replaceConditionalMatch(match: string, template: any): any {
+  match = match.replace(/[\.\\\[\]\(\)]/g, '\\$&');
+
+  const substitute = (toReplace: string) => {
+    return toReplace.replace(/__match/g, match);
+  };
+
+  return recursiveSubstitute(template, substitute, ['__conditions']);
+}
+
+/**
+ * Replaces paths inside "[[]]" with value of the parameter in the path
+ * @param specs - specification of parameter which is being validated
+ * @param rootSpecs - root of the validated specification
+ * @param template - template in which paths will be replaced
+ */
+export function replacePathsWithValues(specs: any, rootSpecs: any, template: any): any {
+  const substitute = (toReplace: string) => {
+    const matches = toReplace.match(/(?<=\[)\[.+?\](?=\])/g);
+
+    if (!matches) {
```

Comment:
When would this happen?

Reply:
This function is called few lines below in `recursiveSubstitute` on every key and value of the specification, it is up to `substitute` to figure out if there is anything to be replaced or not, whatever it returns will replace what was previously there.